pkgs/by-name/cr/crc/package.nix

```diff
@@ -5,74 +5,71 @@
   testers,
   crc,
   coreutils,
+  writableTmpDirAsHomeHook,
 }:
 
 let
-  openShiftVersion = "4.15.12";
+  openShiftVersion = "4.18.2";
   okdVersion = "4.15.0-0.okd-2024-02-23-163410";
-  microshiftVersion = "4.15.12";
-  podmanVersion = "4.4.4";
+  microshiftVersion = "4.18.2";
   writeKey = "$(MODULEPATH)/pkg/crc/segment.WriteKey=cvpHsNcmGCJqVzf6YxrSnVlwFSAZaYtp";
-  gitCommit = "27c493c19b7f396931c3b94cc3367f572e6af04a";
-  gitHash = "sha256-uxp3DVYbbjKf1Cjj7GCf9QBxFq3K136k51eymD0U018=";
+  gitCommit = "e843be9c9889abd33ce2f9aee161fac1d44e3fa8";
+  gitHash = "sha256-irlVpRBZzE6lfjK8nlNmWlryGj25u/5LcX7pG3WD/Fs=";
 in
-buildGoModule rec {
-  version = "2.36.0";
+buildGoModule (finalAttrs: {
   pname = "crc";
+  version = "2.49.0";
 
   src = fetchFromGitHub {
     owner = "crc-org";
     repo = "crc";
-    rev = "v${version}";
+    tag = "v${finalAttrs.version}";
     hash = gitHash;
   };
 
   vendorHash = null;
 
   postPatch = ''
     substituteInPlace pkg/crc/oc/oc_linux_test.go \
-      --replace "/bin/echo" "${coreutils}/bin/echo"
+      --replace-fail "/bin/echo" "${coreutils}/bin/echo"
   '';
 
-  subPackages = [
-    "cmd/crc"
-  ];
+  subPackages = [ "cmd/crc" ];
 
   tags = [ "containers_image_openpgp" ];
 
   ldflags = [
-    "-X github.com/crc-org/crc/v2/pkg/crc/version.crcVersion=${version}"
+    "-X github.com/crc-org/crc/v2/pkg/crc/version.crcVersion=${finalAttrs.version}"
     "-X github.com/crc-org/crc/v2/pkg/crc/version.ocpVersion=${openShiftVersion}"
     "-X github.com/crc-org/crc/v2/pkg/crc/version.okdVersion=${okdVersion}"
-    "-X github.com/crc-org/crc/v2/pkg/crc/version.podmanVersion=${podmanVersion}"
     "-X github.com/crc-org/crc/v2/pkg/crc/version.microshiftVersion=${microshiftVersion}"
     "-X github.com/crc-org/crc/v2/pkg/crc/version.commitSha=${builtins.substring 0 8 gitCommit}"
     "-X github.com/crc-org/crc/v2/pkg/crc/segment.WriteKey=${writeKey}"
   ];
 
-  preCheck = ''
-    export HOME=$(mktemp -d)
-  '';
+  nativeCheckInputs = [ writableTmpDirAsHomeHook ];
 
-  passthru.tests.version = testers.testVersion {
-    package = crc;
-    command = ''
-      export HOME=$(mktemp -d)
-      crc version
-    '';
+  passthru = {
+    updateScript = ./update.sh;
+    tests.version = testers.testVersion {
+      package = crc;
+      command = ''
+        export HOME=$(mktemp -d)
+        crc version
+      '';
```

Comment:
I think it's better to use `versionCheckHook`

Reply:
Use `writableTmpDirAsHomeHook` perhaps?

Reply:
IIRC, using `writableTmpDirAsHomeHook` with `versionCheckHook` is not working because it's not reading environment variables.

What you can do is to alter the command to run and prepend it with `HOME=$(mktemp -d)` ?

Reply:
Should be fixed by https://github.com/NixOS/nixpkgs/pull/403971